src/Servers/Kestrel/Transport.NamedPipes/src/Internal/NamedPipeConnectionListener.cs

```diff
@@ -65,73 +65,79 @@ internal void ReturnStream(NamedPipeServerStream namedPipeServerStream)
 
     public void Start()
     {
-        Debug.Assert(_listeningTasks == null, "Already started");
+        Debug.Assert(_completeListeningTask == null, "Already started");
 
-        _listeningTasks = new Task[_options.ListenerQueueCount];
+        var listeningTasks = new Task[_options.ListenerQueueCount];
 
-        for (var i = 0; i < _listeningTasks.Length; i++)
+        for (var i = 0; i < listeningTasks.Length; i++)
         {
             // Start first stream inline to catch creation errors.
             var initialStream = _namedPipeServerStreamPool.Get();
 
-            _listeningTasks[i] = Task.Run(() => StartAsync(initialStream));
+            listeningTasks[i] = Task.Run(() => StartAsync(initialStream));
         }
+
+        _completeListeningTask = Task.Run(async () =>
+        {
+            try
+            {
+                await Task.WhenAll(listeningTasks);
+                _acceptedQueue.Writer.TryComplete();
+            }
+            catch (Exception ex)
+            {
+                _acceptedQueue.Writer.TryComplete(ex);
```

Comment:
SocketConnectionListener logs the exception and then returns null, it doesn't throw to the caller. Otherwise we'll likely log twice. Should maybe change this to just `TryComplete()` in a `finally`.

Reply:
SocketConnectionListener.AcceptAsync doesn't catch all exceptions. If there is an unexpected exception, then it throws out of the method.

I think what named pipe listener is doing is consistent with that.

Reply:
Oh whoops, I skimmed the code and thought the last catch statement was Exception, but it's SocketException